### PR TITLE
Update InstanceContentOceanFishing

### PR DESCRIPTION
**Pull request by Nukoooo**
- Replaced constructor signature with an absolute one. The old one points to a huge if/else statement, and it will break when SE changes stuff in there.
- Changed the type of ``CurrentZone`` from ``byte`` to ``uint``, let me know if this is a breaking change and I will revert it
- Added 3 structs ``FishDataStruct``, ``AllResultStruct``, ``IndividualResultStruct``

### CODE REVIEW

**Comment by Nukoooo** [comment]
Changed it back to ``byte``, should be good now

**Review by Haselnussbomber — CHANGES_REQUESTED** [review]

File: FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs [hunk 1]
```diff
@@ -1,24 +1,55 @@
 namespace FFXIVClientStructs.FFXIV.Client.Game.InstanceContent;
 
 // ctor (2023.07.06.0000.0000):
-// E9 ?? ?? ?? ?? 3D ?? ?? ?? ?? 0F 87 ?? ?? ?? ?? 3D ?? ?? ?? ?? 0F 83 ?? ?? ?? ?? 05 ?? ?? ?? ?? 83 F8 05
+// 40 53 48 83 EC ?? 48 8B D9 E8 ?? ?? ?? ?? C6 83 ?? ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 03 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 33 C0
 // has id >63000 in InstanceContent sheet
 [StructLayout(LayoutKind.Explicit, Size = 0x1CA8 + 0x650)]
```
Comment (Haselnussbomber) [inline FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs:6]: If you don't mind, I'd like to make two small changes:

1) Having the signatures in quotes allows us to scan it with the `sig_checker.py` script from anna.
2) It allocates `0x2300` bytes for this class, so the size is wrong.

```suggestion
// "40 53 48 83 EC ?? 48 8B D9 E8 ?? ?? ?? ?? C6 83 ?? ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 03 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 48 8D 05 ?? ?? ?? ?? 48 89 83 ?? ?? ?? ?? 33 C0"
// has id >63000 in InstanceContent sheet
[StructLayout(LayoutKind.Explicit, Size = 0x1CA8 + 0x658)]
```

**Review by Haselnussbomber — CHANGES_REQUESTED** [review]

File: FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs [hunk 2]
```diff
@@ -29,4 +60,42 @@ public struct InstanceContentOceanFishing {
...
+    }
+
+    [StructLayout(LayoutKind.Explicit, Size = 0x22)]
+    public struct IndividualResultStruct {
+        [FieldOffset(0x2)] public uint TotalPoints;
+        [FieldOffset(0xA)] public uint ExperiencePoints;
+        // Script introduced in the last expansion
+        [FieldOffset(0xE)] public ushort Script1Amount;
+        // Script that is introduced in this expansion
+        [FieldOffset(0x10)] public ushort Script2Amount;
```
[the remark this answers is not in the document]
Comment (Nukoooo) [reply FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs]: Oops, I should have checked the word in English

File: FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs (hunk 1 above)
Comment (Nukoooo) [reply FFXIVClientStructs/FFXIV/Client/Game/InstanceContent/InstanceContentOceanFishing.cs]: Should be good now

**Comment by Haselnussbomber** [comment]
I think it’s long overdue to merge this.
I asked @wolfcomp a while ago why the requested changes label was added and it was because the struct size wasn‘t matching. This was fixed in the last commit. Probably just forgot to remove the label.
cc: @aers

**Comment by Nukoooo** [comment]
Fixed a conflict, should be good to merge

**Pull request merged**